### PR TITLE
Added a small unit test verifying number of labels = number of models

### PR DESCRIPTION
**Pull request by Buedenbender**
PR is primarily for educational purposes, and also as an addition to issue #8 (last point create a unit test).
I realized that when writing the unit test (confirming _n_ labels = _n_ models) that failing it would not result in any consequences, as the missing labels will just be broadcast. Anyhow, function now warns for missspecified number of labels, and `testthat` includes an additional unit test for that

### CODE REVIEW

**Review by Buedenbender — COMMENTED** [review]
Realized a small mistake, the exemplary_cfa_models.Rdata can be deleted, I currently figuring out how to

**Comment by rempsyc** [comment]
Thanks @Buedenbender ! I will look at this tomorrow.

But usually, I think you should be allowed to make other changes and delete files. However, you must delete the file manually, and the change tracker should see it on your local fork. When you push, it will automatically update here, and the file should be absent. That said, it's also my first time dealing with PRs on my hand, so it could be how I set the permissions...

FYI, on top right you can suggest a reviewer. For educational purposes, I would suggest that you suggest me :P

**Comment by Buedenbender** [comment]
Turns out it was pretty easy to change the PR, but wasnt able to assign a reviewer sorry 
Changes to PR and Reviewers

**Comment by rempsyc** [comment]
> I wasn't able to assign a reviewer

Oops! That's my bad. I got used to the easystats hub, where I am also admin, so probably that's an admin feature! Good to know. I guess usually the maintainer would assign reviewers then.

For deleting the file, I think you made it more complicated then it had to be though. I wonder if it is because I use RStudio interface for version control and you use git. I would like to open a test PR on your package too so we can test it, if you agree.

**Comment by Buedenbender** [comment]
Sure go ahead :)

**Comment by rempsyc** [comment]
Ok so I couldn't figure out how to test locally your PR. There are some resources [[1](https://stackoverflow.com/questions/42982266/how-can-i-test-a-pull-request-before-merging-it)], [[2](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally)], but I don't understand them.

I first tried `create_from_github()` but it doesn't allow me 😦

```r
Error in `stop_bad_github_remote_config()`:
! Unsupported GitHub remote configuration: 'upstream_but_origin_is_not_fork'`
```

So I ended up just using GitHub desktop to copy your repo to a local fork. Then it worked.

**Comment by Buedenbender** [comment]
Thanks for the update, I will check on your PR if I cant get it to work. Could you just add a small change there so that there is something I can see (because currently if I checkout your PR there is no changes).

**Comment by rempsyc** [comment]
It seems that the file was not correctly deleted. Thinks link reads:

> [exemplary_cfa_models.Rdata → tink/exemplary_cfa_models.Rdata](https://github.com/rempsyc/lavaanExtra/pull/9/commits/f1af02a3276aa7a28aa6a556b4eb9f449e3d4168#diff-4e531d90baa5ab97239f037e060fbe6f9789707de468cba605076aa73e51a90c)
File renamed without changes.

Wanna try again? :P

File: R/nice_fit.R
Comment (rempsyc) [reply R/nice_fit.R:43]: I like that it is a warning instead of an error because then it still works if you provide it with a single vector (if for some weird reason you'd like to have the same name for all models), but then it will naturally generate an error if it's too big and also add your warning.

File: tests/testthat/test-nice_fit.R
Comment (rempsyc) [inline tests/testthat/test-nice_fit.R:49]: can you also add an expect error test with a vector that is too large?

**Review by rempsyc — CHANGES_REQUESTED** [review]
Great job! See my inline comments for more good stuff

File: tests/testthat/test-nice_fit.R
Comment (Buedenbender) [reply tests/testthat/test-nice_fit.R:49]: Nice idea! Will implement this 💯

File: R/nice_fit.R
Comment (Buedenbender) [reply R/nice_fit.R:43]: Great catch, I will check this with a Unit Test if this causes a failure and then include a custom error message.

**Comment by Buedenbender** [comment]
I think should be all set, looking forward for your reviews

File: R/nice_fit.R
Comment (Buedenbender) [reply R/nice_fit.R]: Think this should be resolved with the latest commits?

File: tests/testthat/test-nice_fit.R
Comment (Buedenbender) [reply tests/testthat/test-nice_fit.R]: Done by 9941267bab42082c1c4d0a8395ae4a2e31d36b65

**Review by Buedenbender — COMMENTED** [review]
I revised your changes and adapted them with the latest commits. Thanks for the opportunity to experiment with GH tools to collaborative working. 

PS: I am sure that is not a nice thing to do / against etiquette but I accidently reindented your code with [styler](https://styler.r-lib.org/). Guess this is just so in my routine to automatically press the shortcut for reindentation (styler) directly before saving. Hope you don't mind. Actually do you use the styler package (mb just anoter style)?

**Comment by rempsyc** [comment]
> I accidently reindented your code with [styler](https://styler.r-lib.org/).

No worry—I like it ;)

What shortcut do you use? I've only used `style_pkg()` a couple of times, maybe even just in `rempsyc` and not `lavaanExtra`, but formatting my code nicely and for clarity would be desirable.

**Review by rempsyc — APPROVED** [review]

**Comment by rempsyc** [comment]
Thanks for your contribution @Buedenbender ! Now that you are familiar with the procedure, feel free to open other PRs in the future :)

**Comment by rempsyc** [comment]
You can now see yourself in contributors ;D

https://github.com/rempsyc/lavaanExtra/graphs/contributors

**Comment by Buedenbender** [comment]
> > I accidently reindented your code with [styler](https://styler.r-lib.org/).
> 
> No worry—I like it ;)
> 
> What shortcut do you use? I've only used `style_pkg()` a couple of times, maybe even just in `rempsyc` and not `lavaanExtra`, but formatting my code nicely and for clarity would be desirable.

Should be `Ctrl + E`
grafik

**Comment by rempsyc** [comment]
Oh, it wasn't set for me. Thanks! I've set it now :)

styler

**Comment by Buedenbender** [comment]
That was actually quite some fun! Appreciate it :)

**Pull request merged**
